### ISSUE TITLE
Enhancement: [consistent-generic-constructors] Should also fix class properties

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#5285]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-generic-constructors/

### Description

```ts
const x: Map<string, string> = new Map();

export class C {
  public x: Map<string, string> = new Map();
}
```

The rule should autofix both. Currently it only fixes first.

### Fail

```typescript
export class C {
  public x: Map<string, string> = new Map();
}
```

### Pass

```typescript
export class C {
  public x = new Map<string, string>();
}
```
